libraries/AP_Filesystem/examples/File_IO/File_IO.cpp

```diff
@@ -0,0 +1,71 @@
+//Simple File I/O example
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Filesystem/AP_Filesystem.h>
+
+//set according to the data size in file
+#define BUFFER_SIZE 50
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+int open_fd;
+char write_buf[BUFFER_SIZE];
+char read_buf[BUFFER_SIZE];
+char *file_path = nullptr;
+
+void setup();
+void loop();
+
+void setup()
+{
+    //setting path to create a new file
+    file_path = (char*)"./test.txt";
+    hal.console->printf("Creating file - %s\n", file_path);
+
+    //creating or opening a file in write mode
+    open_fd = AP::FS().open(file_path, O_WRONLY | O_CREAT | O_TRUNC);
+    if (open_fd == -1) {
+      hal.console->printf("Open %s failed - %s\n", file_path, strerror(errno));
+      return;
+    }
+
+    //writing Hello World to file
+    char *s = (char*)"Hello World";
+    hal.console->printf("Writing to file => %s\n", s);
+    snprintf(write_buf, sizeof(write_buf), "%s\n", s);
+    const ssize_t to_write = strlen(write_buf);
+    const ssize_t write_size = AP::FS().write(open_fd, write_buf, to_write);
+    if (write_size == -1) {
+      hal.console->printf("Write failed - %s\n", strerror(errno));
+      return;
+    }
+
+    //closing file after writing
+    AP::FS().close(open_fd);
+
+    //opening a file in read mode
+    open_fd = AP::FS().open(file_path, O_RDONLY);
+    if (open_fd == -1) {
+      hal.console->printf("Open %s failed\n", file_path);
+      return;
+    }
+
+    //reading from file
+    const ssize_t read_size = AP::FS().read(open_fd, read_buf, sizeof(read_buf));
+    if (read_size == -1) {
+      hal.console->printf("Read failed - %s\n", strerror(errno));
+      return;
+    }
+
+    hal.console->printf("Reading from file => %s\n", read_buf);
+
+    //closing file after reading
+    AP::FS().close(open_fd);
+}
+
+void loop()
+{
+//empty
```

Comment:
In most examples the main body goes here and is repeated

Reply:
yeah, i saw other examples too, i intend to just run it only once, i can shift the setup() code to loop() but it seems odd to run loop() just once and then exit

Reply:
@Notorious811 actually looping writing consecutive files would be quite reasonable.  A good delay between would be good, with a 1s countdown between iterations.

Problem with just using setup() is that you can lose the entire output of the sketch quite easily.

Reply:
Gotcha, for that should i make those changes in this current PR ? or should i close this & open a new PR